app/models/institution.rb

```diff
@@ -13,11 +13,14 @@ class Institution < ActiveRecord::Base
   has_many :patients, dependent: :destroy
   has_many :samples, dependent: :destroy
   has_many :test_results, dependent: :destroy
+  has_many :roles, dependent: :destroy
 
   validates_presence_of :name
   validates_presence_of :kind
   validates_inclusion_of :kind, in: KINDS
 
+  after_create :create_predefined_roles
```

Comment:
On update, we should update the predefined roles names, if the institution name has changed. Same should apply to sites.